src/api/spec/factories/project.rb

```diff
@@ -11,18 +11,14 @@
 
     factory :project_with_package do
       transient do
-        package_name nil
+        sequence(:package_name) { |n| "package_#{n}" }
         create_patchinfo false
         maintainer nil
       end
 
       after(:create) do |project, evaluator|
-        new_package = if evaluator.package_name
-                        create(:package, project_id: project.id, name: evaluator.package_name)
-                      else
-                        create(:package, project_id: project.id)
-                      end
-        project.packages << new_package
+        package = Package.where(project_id: project.id, name: evaluator.package_name).first
```

Comment:
If I am not mistaken this should always be nil. Since we just created the project